Source/santad/EventProviders/SNTDeviceManager.mm

```diff
@@ -199,43 +200,80 @@ - (void)handleAuthMount:(const es_message_t *)m
     return;
   }
 
-  long mountMode = m->event.mount.statfs->f_flags;
+  struct statfs *eventStatFS;
+  BOOL isRemount = NO;
+
+  switch (m->event_type) {
+    case  ES_EVENT_TYPE_AUTH_MOUNT:
+      eventStatFS = m->event.mount.statfs;
+      break;
+    case ES_EVENT_TYPE_AUTH_REMOUNT:
+      eventStatFS = m->event.remount.statfs;
+      isRemount = YES;
+      break;
+    default:
+      LOGE(@"Unexpected Event Type passed to DeviceManager handleAuthMount: %d", m->event_type);
+      // Fail closed.
+      es_respond_auth_result(self.client, m, ES_AUTH_RESULT_DENY, false);
+      return; 
+    break;
+  }
+
+  long mountMode = eventStatFS->f_flags;
   pid_t pid = audit_token_to_pid(m->process->audit_token);
   LOGI(@"SNTDeviceManager: mount syscall arriving from path: %s, pid: %d, fflags: %lu",
        m->process->executable->path.data, pid, mountMode);
 
   DADiskRef disk =
-    DADiskCreateFromBSDName(NULL, self.diskArbSession, m->event.mount.statfs->f_mntfromname);
+    DADiskCreateFromBSDName(NULL, self.diskArbSession, eventStatFS->f_mntfromname);
   CFAutorelease(disk);
 
   // TODO(tnek): Log all of the other attributes available in diskInfo into a structured log format.
   NSDictionary *diskInfo = CFBridgingRelease(DADiskCopyDescription(disk));
+  BOOL isInternal =  [diskInfo[(__bridge NSString *)kDADiskDescriptionDeviceInternalKey] boolValue];
   BOOL isRemovable = [diskInfo[(__bridge NSString *)kDADiskDescriptionMediaRemovableKey] boolValue];
-  BOOL isUSB =
-    [diskInfo[(__bridge NSString *)kDADiskDescriptionDeviceProtocolKey] isEqualTo:@"USB"];
-
-  if (!isRemovable || !isUSB) {
+  BOOL isEjectable = [diskInfo[(__bridge NSString *)kDADiskDescriptionMediaEjectableKey] boolValue];
+  NSString* protocol = diskInfo[(__bridge NSString *)kDADiskDescriptionDeviceProtocolKey];
+  BOOL isUSB = [protocol isEqualToString:@"USB"];
+  BOOL isDMG = [protocol isEqualToString:@"Virtual Interface"];
+
+  NSString* kind = diskInfo[(__bridge NSString *)kDADiskDescriptionMediaKindKey];
+
+  //TODO: check kind and protocol for banned things (e.g. MTP).
+  LOGD(@"SNTDeviceManager: DiskInfo Protocol: %@", protocol);
+  LOGD(@"SNTDeviceManager: DiskInfo Kind: %@", kind);
+  LOGD(@"SNTDeviceManager: DiskInfo isInternal: %d", isInternal);
+  LOGD(@"SNTDeviceManager: DiskInfo isRemovable: %d", isRemovable);
+  LOGD(@"SNTDeviceManager: DiskInfo isEjectable: %d", isEjectable);
+
+  // If the device isn't a DMG or removable we're ok with the operation.
+  if (isInternal || isDMG || (!isRemovable && !isEjectable && !isUSB)) {
```

Comment:
What is the new condition for?

Reply:
Wanted to make sure we weren't popping up for DMGs explicitly or on internal mounts.